G/Geant4/build_tarballs.jl

```diff
@@ -3,35 +3,55 @@
 using BinaryBuilder
 
 name = "Geant4"
-version = v"10.7.2"
+version = v"11.1.1"
 
 # Collection of sources required to build
 sources = [
-    ArchiveSource("https://github.com/Geant4/geant4/archive/v$(version).tar.gz",
-                  "43a11da475e4b9fd3719a5e589003c1f79c0c6f2cd78bcfe3cea2fe6cd12823b"),
+    ArchiveSource("https://gitlab.cern.ch/geant4/geant4/-/archive/v$(version)/geant4-v$(version).tar.gz",
+                  "c5878634da9ba6765ce35a469b2893044f4a6598aa948733da8436cdbfeef7d2"),
+    ArchiveSource("https://github.com/phracker/MacOSX-SDKs/releases/download/10.15/MacOSX10.15.sdk.tar.xz",
+                  "2408d07df7f324d3beea818585a6d990ba99587c218a3969f924dfcc4de93b62"),
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
 cd $WORKSPACE/srcdir/geant4-*/
+
+if [[ "${target}" == x86_64-apple-darwin* ]]; then
+    # Install a newer SDK which supports `std::filesystem`
+    pushd $WORKSPACE/srcdir/MacOSX10.*.sdk
+    rm -rf /opt/${target}/${target}/sys-root/System
+    cp -ra usr/* "/opt/${target}/${target}/sys-root/usr/."
+    cp -ra System "/opt/${target}/${target}/sys-root/."
+    export MACOSX_DEPLOYMENT_TARGET=10.15
+    popd
+fi
+
 mkdir build && cd build
 FLAGS=()
 if [[ "${target}" != *-apple-* ]]; then
     FLAGS=(-DGEANT4_USE_OPENGL_X11=ON)
 fi
 cmake -DCMAKE_INSTALL_PREFIX=${prefix} \
-    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
-    -DCMAKE_BUILD_TYPE=Release \
-     "${FLAGS[@]}" \
+      -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+      -DCMAKE_BUILD_TYPE=Release \
+      -DCMAKE_CXX_STANDARD=17 \
+      -DGEANT4_INSTALL_DATA=ON \
+      -DGEANT4_USE_GDML=ON \
+      -DGEANT4_BUILD_TLS_MODEL=global-dynamic \
+      "${FLAGS[@]}" \
     ..
 make -j${nproc}
 make install
+
+install_license ../LICENSE
 """
 
 # These are the platforms we will build for by default, unless further
 # platforms are passed in on the command line
-platforms = expand_cxxstring_abis(supported_platforms(; experimental=true))
-filter!(p -> !Sys.iswindows(p) && arch(p) != "armv6l", platforms)
+platforms = expand_cxxstring_abis(supported_platforms())
+platforms = filter(p -> libc(p) != "musl" && os(p) != "windows" && os(p) != "freebsd" && arch(p) != "armv6l" && cxxstring_abi(p) != "cxx03", platforms)
```

Comment:
The [problem with FreeBSD](https://buildkite.com/julialang/yggdrasil/builds/1158#01868f36-d669-46f7-8c22-c355914746dc/6-7884) is
```
[19:36:59] /workspace/srcdir/geant4-v11.1.1/source/global/management/include/tls.hh:72:8: error: "No Thread Local Storage (TLS) technology supported for this platform. Use sequential build !"
```
The error message suggests we should be able to use sequential build (whatever that is) rather thank skipping this platform altogether.

Also, why skipping musl and all C++03 string ABIs?

Reply:
- Yes, the problem with FeeBSD is exactly this, which would mean that I have to configure the build with different options depending on the OS. So, to avoid the extra complexity I decided to remove it from the list of platforms. This is also a platform for which the package Geant4 have never been tested.
- For musl, I was getting errors like:
  ```
/workspace/srcdir/geant4-v11.1.1/source/global/management/include/G4Backtrace.hh:90:12: fatal error: execinfo.h: No such file or directory
   90 | #  include <execinfo.h>
      |            ^~~~~~~~~~~~
compilation terminated.
  ```
  so, since Geant4 was never tested/validated I thought that it was not necessary.
- C++17 is required to build, so I was thinking that was not necessary to support C++03 strings.

Reply:
https://docs.binarybuilder.org/stable/building/#Platforms

> This ABI does *not* have to do with the C++ standard used by the source code, in fact you can build a C++03 library with the C++11 `std::string` ABI and a C++11 library with the C++03 `std::string` ABI. This is achieved by appropriately setting the `_GLIBCXX_USE_CXX11_ABI` macro.

Reply:
Thanks, this is what I was understanding. What I wanted to say is that any 'client' of this library/toolkit needs to be compiled with  -std=c++17, so only the C++11 string will be used in practice.

Reply:
That's ***not*** what I said: the C++ string ABI has ***nothing*** to do with the C++ standard.  You can have a binary with C++03 string ABI built from a C++17 source code.

Reply:
Of course, it does not harm to have them also built. I'll remove it from the filter.